trie/verkle.go

```diff
@@ -216,35 +216,20 @@ func (t *VerkleTrie) RollBackAccount(addr common.Address) error {
 	codeSize := binary.LittleEndian.Uint64(codeSizeBytes)
 
 	// Delete the account header + first 64 slots + first 128 code chunks
-	key := common.CopyBytes(codeSizeKey)
-	for i := 0; i < verkle.NodeWidth; i++ {
-		key[31] = byte(i)
-
-		// this is a workaround to avoid deleting nil leaves, the lib needs to be
-		// fixed to be able to handle that
-		v, err := t.root.Get(key, t.nodeResolver)
-		if err != nil {
-			return fmt.Errorf("error rolling back account header: %w", err)
-		}
-		if len(v) == 0 {
-			continue
-		}
-		_, err = t.root.Delete(key, t.nodeResolver)
-		if err != nil {
-			return fmt.Errorf("error rolling back account header: %w", err)
-		}
+	_, err = t.root.(*verkle.InternalNode).DeleteAtStem(codeSizeKey, t.nodeResolver)
+	if err != nil {
+		return fmt.Errorf("error rolling back account header: %w", err)
 	}
+
 	// Delete all further code
-	for i, chunknr := uint64(32*128), uint64(128); i < codeSize; i, chunknr = i+32, chunknr+1 {
+	for i, chunknr := uint64(31*128), uint64(128); i < codeSize; i, chunknr = i+31*256, chunknr+256 {
```

Comment:
If i understand correctly, the codeSize in the account header refers to the length of *raw contract code*;

After code chunkification, for each chunk we store 1 byte metadata and 31 bytes code? It's the reason you only count 31 bytes here?

Reply:
yes, that is the reason